infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -617,56 +731,78 @@ private void handleAllocation(CorfuPayloadMsg<TokenRequest> msg,
-        Token token = new Token(sequencerEpoch, globalLogTail);
+        Token newToken = new Token(sequencerEpoch, globalLogTail);
         globalLogTail = newTail;
-        r.sendResponse(ctx, msg, CorfuMsgType.TOKEN_RES.payloadMsg(
-                new TokenResponse(token, backPointerMap.build())));
+
+        // Note: we reuse the request header as the ignore_cluster_id and
+        // ignore_epoch fields are the same in both cases.
+        ResponseMsg response = getResponseMsg(
+                req.getHeader(), getTokenResponseMsg(newToken, backPointerMap.build()));
+        r.sendResponse(response, ctx);
     }
 
     /**
      * This method handles the request of streams addresses.
-     *
+     * <p>
      * The request of address spaces can be of two types:
-     *      - For specific streams (and specific ranges for each stream).
-     *      - For all streams (complete range).
-     *
+     * - For specific streams (and specific ranges for each stream).
+     * - For all streams (complete range).
+     * <p>
      * The response contains the requested streams address maps and the global log tail.
      */
-    @ServerHandler(type = CorfuMsgType.STREAMS_ADDRESS_REQUEST)
-    private void handleStreamsAddressRequest(CorfuPayloadMsg<StreamsAddressRequest> msg,
-                                             ChannelHandlerContext ctx, IServerRouter r) {
-        StreamsAddressRequest req = msg.getPayload();
-        Map<UUID, StreamAddressSpace> streamsAddressMap;
-
-        switch (req.getReqType()) {
-            case StreamsAddressRequest.STREAMS:
-                streamsAddressMap = getStreamsAddresses(req.getStreamsRanges());
-                break;
-
-            default:
-                // Retrieve address space for all streams
-                streamsAddressMap = new HashMap<>(this.streamsAddressMap);
-                break;
+    @RequestHandler(type = PayloadCase.STREAMS_ADDRESS_REQUEST)
+    private void handleStreamsAddressRequest(@Nonnull RequestMsg req,
+                                             @Nonnull ChannelHandlerContext ctx,
+                                             @Nonnull IServerRouter r) {
+        StreamsAddressRequestMsg streamsAddressRequest =
+                req.getPayload().getStreamsAddressRequest();
+        Map<UUID, StreamAddressSpace> respStreamsAddressMap;
+
+        if (streamsAddressRequest.getReqType() == StreamsAddressRequestMsg.Type.STREAMS) {
+            respStreamsAddressMap =
+                    getStreamsAddressesMap(streamsAddressRequest.getStreamRangeList());
+        } else if (streamsAddressRequest.getReqType() ==
+                StreamsAddressRequestMsg.Type.ALL_STREAMS) {
+            // Retrieve address space for all streams
+            respStreamsAddressMap = new HashMap<>(streamsAddressMap);
+        } else {
+            throw new IllegalArgumentException("handleStreamsAddressRequest: " +
+                    "Received an INVALID type of streamsAddressRequestMsg.");
```

Comment:
Maybe include the type received as part of the exception message?

Reply:
Added, thanks!